src/client.cpp

```diff
@@ -707,6 +716,43 @@ void CClient::OnControllerInFaderLevel ( int iChannelIdx,
     emit ControllerInFaderLevel ( iChannelIdx, iValue );
 }
 
+void CClient::OnControllerInPanValue ( int iChannelIdx,
+                                       int iValue )
+{
+    // in case of a headless client the panners cannot be moved so we need
+    // to send the controller information directly to the server
+#ifdef HEADLESS
+    // channel index is valid
+    SetRemoteChanPan ( iChannelIdx, static_cast<float>( iValue ) / AUD_MIX_PAN_MAX);
+#endif
+
+    emit ControllerInPanValue ( iChannelIdx, iValue );
+}
+
+void CClient::OnControllerInFaderIsSolo ( int iChannelIdx,
+                                          bool bIsSolo )
+{
+    // in case of a headless client the buttons are not displayed so we need
+    // to send the controller information directly to the server
+#ifdef HEADLESS
+    // FIXME: no idea what to do here.
```

Comment:
I assume we need to implement this before this code can be used?

Reply:
Depends on what you mean by "can be used".  The code compiles fine in headless mode.  As far as I understand (which may perfectly well be wrong), the headless client does not entertain notions of mute and solo state (which are derived as the state of GUI toggle buttons in the GUI client) and one would have to invent data structures for keeping track of them.

That's nothing I am up to.  At the same time, the code compiles and does not work worse than before.  I don't quite see the point in removing support for mute/solo when using a GUI in order to hobble it as much as headless mode.

Reply:
I think it was "just done that way" rather than try to keep two data structures synchronised - i.e. the client UI and the client core.  Currently, in many places, the client UI tells the client core to update but all the client core does is tell the server to update.